File: android/src/main/kotlin/me/sithiramunasinghe/flutter/flutter_radio_player/core/events/FRPPlayerEvent.kt
package me.sithiramunasinghe.flutter.flutter_radio_player.core.events

import com.google.gson.annotations.SerializedName
import me.sithiramunasinghe.flutter.flutter_radio_player.core.data.FRPCurrentSource
import me.sithiramunasinghe.flutter.flutter_radio_player.core.data.FRPVolumeChangeEvent

data class FRPPlayerEvent(
    @SerializedName("type")
    val type: String? = null,

    @SerializedName("currentSource")
    val currentSource: FRPCurrentSource? = null,

    @SerializedName("volumeChangeEvent")
    val volumeChangeEvent: FRPVolumeChangeEvent? = null,

    @SerializedName("playbackStatus")
    val playbackStatus: String? = null,

    @SerializedName("icyMetaDetails")
    val icyMetaDetails: String? = null
)

File: ios/Classes/core/data/FRPPlayerEvent.swift
//
//  FRPPlayerEvent.swift
//  flutter_radio_player
//
//  Created by Sithira Munasinghe on 2022-05-25.
//

import Foundation

struct FRPPlayerEvent: Codable {
    var currentSource: FRPCurrentSource? = nil
    var volumeChangeEvent: FRPVolumeChangeEvent? = nil
    var type: String? = nil
    var playbackStatus: String? = nil
    var icyMetaDetails: String? = nil
}
